sync/awsdatabricks.py

```diff
@@ -623,16 +703,71 @@ def _get_task_cluster(task: dict, clusters: list) -> Response[dict]:
     return Response(result=cluster)
 
 
-def _get_eventlog(task: dict, cluster: dict) -> Response[bytes]:
+def _s3_contents_have_all_rollover_logs(contents: list[dict], run_end_time_seconds: float):
+    final_rollover_log = contents and next(
+        (
+            content
+            for content in contents
+            if Path(content["Key"]).stem in {"eventlog", "eventlog.gz", "eventlog.json.gz"}
+        ),
+        False,
+    )
+    return (
+        # Related to - https://docs.databricks.com/clusters/configure.html#cluster-log-delivery-1
+        # We want to make sure that the final_rollover_log has had data written to it AFTER the run has actually
+        # completed. We use this as a signal that Databricks has flushed all event log data to S3, and that we
+        # can proceed with the upload + prediction. If this proves to be finnicky in some way, we can always
+        # wait for the full 5 minutes after the run_end_time_seconds
+        final_rollover_log
+        and final_rollover_log["LastModified"].timestamp() >= run_end_time_seconds
+    )
+
+
+def event_log_poll_duration_seconds():
+    """Convenience function to aid testing"""
+    return 15
+
+
+def _get_eventlog(task: dict, cluster: dict, run_end_time_millis: int) -> Response[bytes]:
     log_url = cluster.get("cluster_log_conf", {}).get("s3", {}).get("destination")
     if log_url:
         parsed_log_url = urlparse(log_url)
-
+        cluster_id = task.get("cluster_instance", {}).get("cluster_id")
```

Comment:
True - or just pass in the known cluster ID that we check for in `create_prediction_for_run`. We already do the cluster ID error handling there, and we don't use the `task` argument for anything else in this function, by my eye. Since this is a private function, I think requiring a non-`None` argument is reasonable.